klippa/src/glyf_loca.rs

```diff
@@ -0,0 +1,304 @@
+    Ok(())
+}
+
+fn padded_size(len: usize) -> usize {
+    len + len % 2
+}
+
+fn subset_glyph(glyph: &Glyph, plan: &Plan) -> Vec<u8> {
+    //TODO: support set_overlaps_flag and drop_hints
+    match glyph {
+        Composite(comp_g) => subset_composite_glyph(comp_g, plan),
+        Simple(simple_g) => subset_simple_glyph(simple_g),
+    }
+}
+
+// TODO: drop_hints and set_overlaps_flag
+fn subset_simple_glyph(g: &SimpleGlyph) -> Vec<u8> {
+    let mut out = Vec::with_capacity(g.offset_data().len());
+
+    let Some(num_coords) = g.end_pts_of_contours().last() else {
+        return out;
+    };
+    let num_coords = num_coords.get() + 1;
+    let glyph_data = g.glyph_data();
+    let i = trim_simple_glyph_padding(glyph_data, num_coords);
+    if i == 0 {
+        return out;
+    }
+    let trimmed_len =
+        10 + 2 * (g.number_of_contours() as usize) + 2 + g.instruction_length() as usize + i;
+    let Some(trimmed_slice) = g.offset_data().as_bytes().get(0..trimmed_len) else {
+        return out;
+    };
+    out.extend_from_slice(trimmed_slice);
+    out
+}
+
+fn subset_composite_glyph(g: &CompositeGlyph, plan: &Plan) -> Vec<u8> {
+    let mut out = Vec::with_capacity(g.offset_data().len());
+    out.extend_from_slice(g.offset_data().as_bytes());
+
+    let mut more = true;
+    let mut we_have_instructions = false;
+    let mut i: usize = 10;
+    let len: usize = out.len();
+
+    while more {
+        if i + 3 >= len {
+            return Vec::new();
+        }
+        let flags = u16::from_be_bytes([out[i], out[i + 1]]);
+        let flags = CompositeGlyphFlags::from_bits_truncate(flags);
+
+        if flags.contains(CompositeGlyphFlags::WE_HAVE_INSTRUCTIONS) {
+            we_have_instructions = true;
+        }
+
+        let old_gid = u16::from_be_bytes([out[i + 2], out[i + 3]]);
+        let Some(new_gid) = plan.glyph_map.get(&GlyphId::from(old_gid)) else {
+            return Vec::new();
+        };
+        let new_gid = new_gid.to_u32() as u16;
+        out[i + 2] = (new_gid >> 8) as u8;
+        out[i + 3] = (new_gid & 0xFF) as u8;
+
+        i += 4;
+
+        if flags.contains(CompositeGlyphFlags::ARG_1_AND_2_ARE_WORDS) {
+            i += 4;
+        } else {
+            i += 2;
+        }
+
+        if flags.contains(CompositeGlyphFlags::WE_HAVE_A_SCALE) {
+            i += 2;
+        } else if flags.contains(CompositeGlyphFlags::WE_HAVE_AN_X_AND_Y_SCALE) {
+            i += 4;
+        } else if flags.contains(CompositeGlyphFlags::WE_HAVE_A_TWO_BY_TWO) {
+            i += 8;
+        }
+
+        more = flags.contains(CompositeGlyphFlags::MORE_COMPONENTS);
+    }
+
+    if we_have_instructions {
+        if i + 1 >= len {
+            return Vec::new();
+        }
+        let instruction_len = u16::from_be_bytes([out[i], out[i + 1]]);
+        i += 2 + instruction_len as usize;
+    }
+
+    out.truncate(i);
+    out
+}
+
+fn trim_simple_glyph_padding(glyph_data: &[u8], num_coords: u16) -> usize {
```

Comment:
I think this deserves a bit of explanation, it isn't clear to me what padding we're expecting to find? is it only the padding required to make the offset an even number?

I'd also document the meaning of the return value.